cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -187,21 +189,33 @@ public String getPartition(PartitionKey key) {
         if (row == null) {
           break;
         }
-        if (filter != null && !validateFilter(filter, row)) {
+        PartitionKey key;
+        try {
+          key = parseRowKey(row.getRow(), partitioning);
+        } catch (IllegalArgumentException e) {
+          if (!ignoreInvalidRowsSilently) {
```

Comment:
so ignore silently just means we don't debug log but still ignore the row?  Is it expected that we won't be able to parse the row key?